src/tdastro/sources/physical_model.py

```diff
@@ -116,6 +116,49 @@ def set_graph_positions(self, seen_nodes=None):
         for effect in self.effects:
             effect.set_graph_positions(seen_nodes=seen_nodes)
 
+    def _redshift_pre_effect(self, observer_frame_times, observer_frame_wavelengths, params):
+        """Calculate the rest frame times and wavelengths needed to give user the observer frame times
+        and wavelengths (given the redshift).
+
+        Parameters
+        ----------
+        observer_frame_times : numpy.ndarray
+            The times at which the observation is made.
+        observer_frame_wavelengths : numpy.ndarray
+            The wavelengths at which the observation is made (in Angstroms).
+        params : `dict`
```

Comment:
I have a slight preference for passing the redshift and t0 explicitly so this function doesn't need to worry about how the dictionary is stored. The reason is that it lets users know exactly what is expected by the function.

If you'd prefer to keep it as-is (passing in `params`), you should add a note to the doc string that params needs to include redshift and t0.

Reply:
Now directly passing redshift and t0 to `obs_frame_to_rest_frame` and redshift to `apply_redshift`